sphinx/util/inspect.py

```diff
@@ -116,6 +116,15 @@ def getargspec(func: Callable) -> Any:
                                kwonlyargs, kwdefaults, annotations)
 
 
+def unwrap(obj: Any) -> Any:
+    """Get an original object from wrapped object (wrapped functions)."""
+    try:
+        return inspect.unwrap(obj)
```

Comment:
How about providing some callback to `unwrap` to check if wrapped object (if there is any) id is equal to the object itself. Like (based on https://github.com/python/cpython/blob/master/Lib/inspect.py#L509):
```
def checker(f):
     return id(f) == id(f.__wrapped__)
inspect.unwrap(obj, stop = checker)
```
I guess that you may get `ValueError` for an object that is not mock and miss the real error.

Reply:
@JanuszL `ValueError` is being thrown from https://github.com/python/cpython/blob/master/Lib/inspect.py#L524 and it's catching exactly the error you're looking for this to catch.

see: https://github.com/python/cpython/blob/master/Lib/inspect.py#L506

Reply:
Yes, ValueError means "a cycle is encountered". It's a real error in this case. I think it is an abnormal case. So let's talk if we encountered.

Reply:
Sure. I was only thinking about the situation where the cycle is encountered and this is not due to mocking. In such a case it should really throw, so maybe there is an easier way to check if this is a moc or some valid object.